### DISCUSSION TITLE
Should console command 'resave' update the dateUpdated (at least optionally?)

### DISCUSSION THREAD

**Discussion by bossanova808** [craftcms/cms#11849] [Ideas] [answered]
So in trying to correct some SEO stuff on my site, which uses SEOmatic by @khalwat, I found myself wanting to trigger a mass updating of the Last Modified date of a bunch of products, as visible on the generated Sitemap (see e.g. https://imagescience.com.au/sitemaps-1-section-brand-1-sitemap.xml).

When I manually resave one of those products from the UI, the date updated is updated as expected (although also worth noting that there is no UI to show this in Products as there is in Entries...and I think there should be...) - and this is immediately reflected on the generated sitemap as one would hope.

So I turned to `./craft resave/products` as the quick method of 'touching' all the product dates.  Only to find this does _not_ work.  I can see the rows in the `craft_products` and `craft_elements` tables _do_ get updated, but clearly that is not that actual data used for the latest update record (I think maybe the values used are in `craft_content`?)

I am sure there are good reasons why a `resave` by default leaves the modified date intact, but it seems to be it would be useful if there WAS a way to manually 'touch' elements on resave.  Perhaps

`./craft resave/products --touch=true` or similar?

I did see the `--set` `--to` options, but have not attempted that - date formats on command lines always make nervous (do I use a PHP date format, or mySQL or...?), and I am not sure these 'special attributes' would actually get updated anyway.  And Andrew, looking at the code, concluded that `--set myCustomField --to "someValue"` would appear to result in the `dateUpdated` not being updated, either, as it's going through the resave code which seems to expressly preclude it from being updated.

So all in all - any chance of `--touch`?  And while I am asking for unreasonable things...back ported to Craft 3 too?

**Comment by khalwat** [comment]
One comment I had whilst looking into this for @bossanova808 was that if you did something like:

```
./craft resave/products --set myCustomField --to "someValue"
```
...where element content is actually being changed, that `dateUpdated` for the element probably should be updated as well, but from my reading of the code, if elements are being resaved, it specifically does not updated `dateUpdated` no matter what.

**Comment by brandonkelly** [comment, marked as answer]
Added a `--touch` option for the next Craft 3 and 4 releases! (48f795040d2d7d284a8027b281718afde84105c8)

**Reply by bossanova808** [reply]
Fantastic, Brandon, many thanks!

**Reply by khalwat** [reply]
Very cool!

**Reply by brandonkelly** [reply]
Craft 3.7.54 and 4.2.4 are now tagged!

**Reply by bossanova808** [reply]
@brandonkelly This worked a treat for my purposes, on first try - thanks again!
